packages/big-design/src/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,103 @@
+import { ArrowDropDownIcon, ChevronLeftIcon, ChevronRightIcon } from '@bigcommerce/big-design-icons';
+import React, { useEffect, useState } from 'react';
+
+import { MarginProps } from '../../mixins';
+import { Dropdown } from '../Dropdown';
+import { Flex } from '../Flex';
+
+import { StyledButton } from './styled';
+
+export interface PaginationProps extends MarginProps {
+  currentPerPage: number;
+  currentPage: number;
+  numOfItems: number;
+  perPageOptions: number[];
+  onPageUpdate(page: number): void;
+  onRangeUpdate(range: number): void;
```

Comment:
Talking to @chanceaclark it seems there's an opportunity to make some of this `prop` names descriptive, since they were a little confusing to him first.

Some suggestions would be to change `onPageUpdate` & `onRangeUpdate` to `onPageChange` & `onRangeChange`.

I also think we need to match `currentPerPage` to `onRangeUpdate` and change it to `currentRange`. Or vice versa.

Another suggestion would be to change `numOfItems` to `maxItems`.